provider/blob/src/blob_data_provider.rs

```diff
@@ -68,17 +69,34 @@ impl BlobDataProvider {
                 BlobSchema::deserialize(&mut postcard::Deserializer::from_bytes(bytes)).map(
                     |blob| {
                         let BlobSchema::V001(blob) = blob;
-                        blob.resources
+                        blob
```

Comment:
You're adding new invariants to `BlobSchemaV1`, which I think we should verify here. Something like
```
blob.keys.iter_values().max() < blob.buffers.len()
```

Reply:
We could check the invariant upon construction, but it means we need to scan over the buffer. I'd rather lean towards GIGO behavior than having a somewhat expensive validation operation for a case that should never happen.

I will make it a debug assertion instead.